Dockerfile

```diff
@@ -0,0 +1,32 @@
+# Copyright 2021 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# About: This dockerfile builds the cip binary for use in auditor e2e tests.
+
+FROM golang:1.16 AS cip-bin
```

Comment:
I tagged golang  `v1.16` with stability in mind. If we're more concerned with reducing build maintenance, I'm happy to tag `latest`. I do call `go_with_version.sh` on line 22 (ensuring the build info is stamped), but don't feel it should replace building within the container (seems like Docker's job). If anything, using the `latest` tag will ensure the auditor is supported by future versions of go, since this logic is guarded by auditor tests.